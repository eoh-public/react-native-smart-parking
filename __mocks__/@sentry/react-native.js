export const setUser = jest.fn();
export const init = jest.fn();

export default {
  init,
  setUser,
};
