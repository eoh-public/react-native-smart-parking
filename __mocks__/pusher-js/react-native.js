const channel = {
  bind: jest.fn(),
};

const pusherClient = {
  subscribe: jest.fn(),
  unsubscribe: jest.fn(),
  disconnect: jest.fn(),
};

pusherClient.subscribe.mockImplementation(() => channel);

const MockPusher = jest.fn();
MockPusher.mockImplementation(() => pusherClient);

export default MockPusher;
