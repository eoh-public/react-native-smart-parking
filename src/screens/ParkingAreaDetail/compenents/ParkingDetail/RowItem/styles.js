import { StyleSheet } from 'react-native';
import { Device } from '../../../../../configs';

export const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingVertical: 8,
    paddingLeft: 16,
  },
  img: {
    width: 20,
    height: 20,
    marginLeft: 16,
  },
  txtTitle: {
    marginTop: !Device.isIOS ? 4 : 0,
    marginLeft: 18,
  },
});
