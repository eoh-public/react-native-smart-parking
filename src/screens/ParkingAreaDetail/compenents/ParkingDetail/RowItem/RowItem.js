import React, { memo } from 'react';
import { View } from 'react-native';

import Text from '../../../../../commons/Text';

import { styles } from './styles';

const RowItem = memo(({ source, title, subTitle, titleColor }) => {
  return (
    <View style={styles.container}>
      {source}
      <Text type="H4" style={styles.txtTitle} color={titleColor}>
        {title}
        {subTitle && <Text> {subTitle} </Text>}
      </Text>
    </View>
  );
});

export default RowItem;
