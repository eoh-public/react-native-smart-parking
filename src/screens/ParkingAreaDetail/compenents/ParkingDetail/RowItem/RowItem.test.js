import React from 'react';
import { act, create } from 'react-test-renderer';
import RowItem from './RowItem';
import { View } from 'react-native';

describe('Test RowItem', () => {
  let tree;
  test('render row item', async () => {
    const title = 'title';
    act(() => {
      tree = create(<RowItem source={<View />} title={title} />);
    });

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
