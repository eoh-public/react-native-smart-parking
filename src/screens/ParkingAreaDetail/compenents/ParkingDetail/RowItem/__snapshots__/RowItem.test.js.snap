// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RowItem  render row item 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "paddingLeft": 16,
      "paddingVertical": 8,
    }
  }
>
  <View />
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 16,
          "lineHeight": 24,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginLeft": 18,
          "marginTop": 0,
        },
      ]
    }
  >
    title
  </Text>
</View>
`;
